src/routers/alpha-router/alpha-router.ts

```diff
@@ -1885,6 +1900,31 @@ export class AlphaRouter
     return optimalRatio;
   }
 
+  private async userHasSufficientBalance(
+    fromAddress: string,
+    tradeType: TradeType,
+    amount: CurrencyAmount,
+    quote: CurrencyAmount
+  ): Promise<boolean> {
+    try {
+      const neededBalance = tradeType == TradeType.EXACT_INPUT ? amount : quote;
+      let balance;
+      if (neededBalance.currency.isNative) {
+        balance = await this.provider.getBalance(fromAddress);
+      } else {
+        const tokenContract = Erc20__factory.connect(
+          neededBalance.currency.address,
+          this.provider
+        );
+        balance = await tokenContract.balanceOf(fromAddress);
+      }
+      return balance.gte(BigNumber.from(neededBalance.quotient.toString()));
+    } catch (e) {
+      log.info(e, 'Error while checking user balance');
```

Comment:
```suggestion
      log.error(e, 'Error while checking user balance');
```
Should we log this as an error instead?